example/src/components/formExample.js

```diff
@@ -18,7 +19,7 @@ export default React.createClass({
 
     getInitialState() {
         return {
-            showCode: false,
+            showCode: this.props.showCode || false,
```

Comment:
Excellent question.

The use for this prop is to decide if the code or the result should be shown first by default. With demos people are excited to see moving parts, so showing results is best. When building up a guide, we talk in code not components. It is better for people to see code first and then see the result. Perhaps the naming is poor or I should describe this on the `propTypes`.

In regards to the anti pattern, I totally agree. In general, using props in `getInitialState`  is duplication and bad.

> However, it's not an anti-pattern if you make it clear that synchronization's not the goal here

The last section of that page is what is going on here. `showCode` is internal state to the component that will never change based on props. Therefore synchronization is not an issue here. This does draw to light the fact that these variables have :hankey: names. The fact that they are named the same implies that they are duplicates. 

With this in mind, I will change the prop name to `showCodeByDefault` to make it more explicit that duplication is not happening.